lms/djangoapps/labster_vouchers/views.py

```diff
@@ -66,10 +66,54 @@ def enter_voucher(request):
     return render_to_response('labster/enter_voucher.html', context)
 
 
+def activate_voucher(voucher, user_id, email, context_id):
+    """
+    Activates the voucher.
+    """
+    url = settings.LABSTER_ENDPOINTS.get('voucher_activate')
+    headers = {
+        "authorization": 'Token {}'.format(settings.LABSTER_API_AUTH_TOKEN),
+        "accept": 'application/json',
+    }
+
+    data = {
+        'user_id': user_id,
+        'email': email,
+        'context_id': context_id,
+        'voucher': voucher,
+    }
+
+    response = None
+    try:
+        response = requests.post(url, headers=headers, data=data)
+        response.raise_for_status()
+    except RequestException as ex:
+        if getattr(response, 'status_code', None) == 404:
+            raise ItemNotFoundError
+        else:
+            msg = (
+                "Issues with access code activation: user_id='%s', email='%s', "
+                "context_id='%s', voucher='%s',\nerror:\n%r"
+            )
+            log.exception(msg, user_id, email, context_id, voucher, ex)
+            raise LabsterApiError(_("Labster API is unavailable."))
```

Comment:
Why not exception message? At least for debugging purpose.

Reply:
> Why not exception message? At least for debugging purpose.

It makes no sense because the expaction is catched in the parent view. That's why there is a `log.exception` above for debugging purposes.